core/tauri/src/window.rs

```diff
@@ -1648,8 +1648,14 @@ impl<R: Runtime> Window<R> {
     self.current_url.clone()
   }
 
+  /// Navigates the webview to the defined url.
+  #[cfg(not(test))]
+  pub fn navigate(&mut self, url: Url) {
+    self.window.dispatcher.navigate(url).unwrap();
+  }
+
   #[cfg(test)]
-  pub(crate) fn navigate(&mut self, url: Url) {
+  pub fn navigate(&mut self, url: Url) {
     self.current_url = url;
```

Comment:
Now that we have an actual `navigate` function, we can remove the [current_url field](https://github.com/tauri-apps/tauri/blob/1542ad17d647460ec7e2750182a502debc110e28/core/tauri/src/window.rs#L782) and implement `fn navigate` under the [mock runtime](https://github.com/tauri-apps/tauri/blob/1542ad17d647460ec7e2750182a502debc110e28/core/tauri/src/test/mock_runtime.rs#L553). Basically you'll need to change the [Dispatcher#url](https://github.com/tauri-apps/tauri/blob/1542ad17d647460ec7e2750182a502debc110e28/core/tauri/src/test/mock_runtime.rs#L193) to a `Mutex<String>` and mutate it on the `navigate` impl.

With this change we have less `#[cfg(test)] #[cfg(not(test))]` to deal with.

Reply:
I'll get that added.

Reply:
I've reworked how url is handled in test contexts.